ios/Podfile.lock

```diff
@@ -1,6 +1,18 @@
 PODS:
   - boost-for-react-native (1.63.0)
   - DoubleConversion (1.1.6)
+  - EXApplication (2.1.0):
+    - UMCore
```

Comment:
IIUC how this works, I think this part should happen only in the second commit.

(In this instance, the main value of getting this right is probably that it helps this commit serve as a clean demonstration of how unimodules work! But also I think it avoids having the first commit be broken. So it's also an example of how "every commit keeps things working" helps with "each commit is understandable in itself".)

Reply:
Hmm, it looks like you've selected lines 4 and 5:

```diff
+   - EXApplication (2.1.0):
+     - UMCore
```

and these lines were added only in the second commit, if I'm reading my graphical Git client and the GitHub UI correctly. 🙂 

Or are you referring to all of these additions to the Podfile.lock, e.g., EXAppLoaderProvider, EXConstants, EXFileSystem, EXPermissions, and a bunch of things prefixed with UM? If so:

It looks like the `use_unimodules` script called from the Podfile (it's at node_modules/react-native-unimodules/cocoapods.rb) goes through node_modules and, for any module that has a `unimodule.json` file in it, includes it with `pod "#{pod_name}", path: podspec_directory` (the same way other pods are included in the Podfile), which explains how these lines are getting added in the Podfile.lock. So, after running Yarn at this commit, to make sure node_modules is in the correct state, we'll want to be sure we know the reason why any module with a `unimodule.json` is in node_modules.

We can take exactly the same input, or nearly — just those `unimodule.json` files, without looking elsewhere in a given module — and ask Yarn why they're there:

`for f in node_modules/*/unimodule.json; do yarn why $(cat $f | jq -r '.name'); done` (I'd appreciate feedback on this little script, if you have any 🙂)

and all of them tell me the same story, namely:

```
info Reasons this module exists
   - "react-native-unimodules" depends on it
```

And 'react-native-unimodules' is, correctly, the only new dependency we're adding to the `package.json` in this commit.

Reply:
> and these lines were added only in the second commit, if I'm reading my graphical Git client and the GitHub UI correctly. :slightly_smiling_face: 

Hmm! Indeed they were. Sorry about that -- I misread the branch. Those lines (and the other `EXApplication`-related lines in this file) were the ones I had in mind.

I think I was actually fooled by the `Podfile.lock -diff` attribute setting which I asked for on the previous PR <_< >_>... or perhaps by the combination of that and the fact that I have `yarn.lock diff` in my `.git/info/exclude`, causing `yarn.lock` and `Podfile.lock` to behave differently in the diffs I see.

So in the second commit (which is now the third commit -- the one adding `expo-application`), I saw it appearing in `yarn.lock`, meaning it wasn't in `node_modules/` before that point... but didn't see any diff in the `Podfile.lock` -- there would have been a block like this:
```diff
diff --git ios/Podfile.lock ios/Podfile.lock
index 24f59e6bd..3c581ef7b 100644
Binary files ios/Podfile.lock and ios/Podfile.lock differ
```
but that's easy to skim past -- and inferred it was already added in the first commit.

I think the immediate lesson is I'll be sure to keep those two files either both at our default of `-diff`, or both overriden to `diff`, in my `.git/info/exclude` so I don't see a spurious mismatch between them.

Reply:
No problem! 😄